### ISSUE TITLE
Double Scrobbles

### ISSUE THREAD

**Issue by jonappleseed** [rurza/NepTunes-Issues#7]
Using Neptunes 1.2 (2889) on Mac 10.11.3 and I keep getting double scrobbles for any tracks played through iTunes.

**Comment by rurza** [comment]
look for different scrobblers turned on, because it's not technically possible for NepTunes to have double scrobbles. You can always turn on debugMode in Terminal.app (you can copy-paste code below)
`defaults write pl.micropixels.NepTunes debugMode YES`

<img width="682" alt="screen shot 2016-03-14 at 14 03 19" src="https://cloud.githubusercontent.com/assets/2467137/13745277/832b0858-e9ed-11e5-9b9f-aa5493db34df.png">
NepTunes will print everything it does in Console.app (both Terminal and Console are built-in applications) like on screenshot : 
<img width="920" alt="screen shot 2016-03-14 at 14 23 05" src="https://cloud.githubusercontent.com/assets/2467137/13745794/488447ac-e9f0-11e5-8105-03d75edb52e7.png">

To turn off debug mode paste code below to the Terminal:
`defaults write pl.micropixels.NepTunes debugMode NO`

**Comment by rurza** [comment]
Also, it is possible, that Last.fm itself doubles your scrobbles :( Unfortunately I can't help with that, but they have [support](https://getsatisfaction.com/lastfm).

**Comment by hostep** [comment]
I have noticed this too lately, but it isn't a big issue.
It only happens while playing a song, then when you look at your last.fm profile, you'll see the current song displayed twice. But when the next song start, the duplicate of the previous song is removed.
So I don't think this is a big problem.

**Comment by rurza** [comment]
:) This behavior is completely normal :)
After 5s of new track NepTunes sends "now playing" info to Last.fm. After half of the duration of the track (or later, it can be changed in preferences) it sends that track was "scrobbled". So you can see two of them on Last.fm profile page but actually only one is scrobbled.
It's related to old Last.fm website where it was more obvious :)
I'm not going to change it but thanks for your info anyway!

**Comment by jonappleseed** [comment]
I have to disagree because 9 out of 10 songs I've played are all scrobbled twice, not just the current song that is playing. I've been going to my profile page and removing them manually.

**Comment by rurza** [comment]
@jonappleseed Did you turn on "debug mode" in NepTunes? What does it say? I'm still convinced that you have two scrobblers turned on because, like I wrote earlier it isn't possible for NepTunes to doubles scrobble music **by design**. You can always check for them in Activity Monitor.app:

<img width="922" alt="screen shot 2016-03-17 at 06 55 34" src="https://cloud.githubusercontent.com/assets/2467137/13837613/469c99de-ec0d-11e5-85f7-389aa2dde6e4.png">

The thing that @hostep wrote about is unrelated.

<img width="650" alt="screen shot 2016-03-17 at 06 21 29" src="https://cloud.githubusercontent.com/assets/2467137/13837153/857dc9ca-ec08-11e5-995c-01645eee65d1.png">

<img width="657" alt="screen shot 2016-03-17 at 06 22 10" src="https://cloud.githubusercontent.com/assets/2467137/13837159/9b2f8d44-ec08-11e5-98f8-53ed2ed0202e.png">

By "scrobbling now" Last.fm means "You're listening".
By "scrobbled" Last.fm means "You've listened it and I've saved it"
With next track everything is fine:
<img width="667" alt="screen shot 2016-03-17 at 06 23 20" src="https://cloud.githubusercontent.com/assets/2467137/13837191/ec7aa9ae-ec08-11e5-9682-217955aecfca.png">

This screenshot I've found on Google:
![mediamonkey-scrobbling-2a_zps35f40934](https://cloud.githubusercontent.com/assets/2467137/13837217/36f6f64a-ec09-11e5-91ed-26ddf9890fae.JPG)
It's from the old website where, like I wrote in previous post it was more obvious.

**Comment by jonappleseed** [comment]
My console doesn't seem to be reporting any errors actually and I'm certain I do not have any other scrobblers active. 

![screen shot 2016-03-17 at 2 13 16 pm](https://cloud.githubusercontent.com/assets/17821937/13838688/ecadc53a-ec50-11e5-826d-1993f7b62d26.png)
![screen shot 2016-03-17 at 2 58 12 pm](https://cloud.githubusercontent.com/assets/17821937/13838691/f2942598-ec50-11e5-9a82-6fcd272d83ee.png)

**Comment by rurza** [comment]
hm...
Note that there is a ~2 minutes difference between scrobbles.
I must ask, did you used any scrobbler before NepTunes? 
You know, a lot of them has an option to hide icons and can start with iTunes itself (or OS X) so it's super easy to forgot about them. 

What I would do? 
I would download [Little Snitch](https://www.obdev.at/products/littlesnitch/index.html).
This app informs you when something sends data from your computer to the Internet (any scrobbler does that). It's a paid app but it has a demo version that works like a normal version.

![screen shot 2016-03-17 at 08 32 53](https://cloud.githubusercontent.com/assets/2467137/13839384/f4d04d36-ec1a-11e5-93d3-06a8c17c9ba7.png)

If this doesn't help send me your username and I will write to Last.fm for you :)

**Comment by jonappleseed** [comment]
I did use Lyrica before Neptunes but have since uninstalled that using AppDelete and don't see any instance of that in my Activity Monitor nor Login Items. I've installed Little Snitch and it's not reporting any scrobbler app other than Neptunes sending out last.fm connections. 

My Last.fm ID is icubes. Thank you so much.

**Comment by rurza** [comment]
Also, you can always reset NepTunes. Quit NepTunes, remove everything in:
`~/Library/Containers/pl.micropixels.NepTunes/`
You can go to this folder by pressing CMD+SHIFT+G in Finder:)
![screen shot 2016-03-17 at 10 21 21](https://cloud.githubusercontent.com/assets/2467137/13841590/238872a2-ec2a-11e5-8523-82e4c8ab32ef.png)

And I wrote to Last.fm support.

**Comment by rurza** [comment]
@jonappleseed 
![screen shot 2016-03-17 at 18 10 27](https://cloud.githubusercontent.com/assets/2467137/13854503/16974158-ec6c-11e5-8675-930ea5f01145.png)

**Comment by jonappleseed** [comment]
Thank you!! Hope it gets resolved soon

**Comment by rurza** [comment]
@jonappleseed reset of settings didn't help? :(

**Comment by jonappleseed** [comment]
I will try that once I get back. Will keep you posted

**Comment by rurza** [comment]
@jonappleseed Stll no answer from Last.fm support. Anything changed on your side?

**Comment by rurza** [comment]
I have theory that on some accounts Last.fm treats "now playing" as a scrobbled. You can help with this. To confirm my theory (and if you have doubled scrobbles) play a track and after ~10 seconds go to your Last.fm profile. If my theory is right you should have this track already scrobbled!

**Comment by jonappleseed** [comment]
Hey Adam, still getting doubled scrobbles. I have to constantly go to my last.fm page to delete them manually. Not sure how we can resolve the 'now playing' bug? Am I the only one experiencing this??

**Comment by rurza** [comment]
@jonappleseed 
:( Within one year I had maybe 5 incidents - 3 of them were caused by a second scrobbler turned on, 1 didn't answered and you. Last.fm didn't answered to me :/ 
I have another idea: Can you create new account and test NepTunes with it? You know, even for 15 minutes.
We can try to resolve this issue by special version of NepTunes just for you - without "now playing" feature :D
I'm at work right now so I can't help you right now, but send me an e-mail: adam@micropixels.pl, I will send you new version (1.30.1 aka 1.30-jonappleseed) later.

![screen shot 2016-04-04 at 08 48 15](https://cloud.githubusercontent.com/assets/2467137/14240057/fae20acc-fa41-11e5-9847-8384a7172762.png)

**Comment by rurza** [comment]
@jonappleseed  Something changed with new version?

**Comment by jonappleseed** [comment]
Hey Adam, unfortunately still seeing the same double scrobbles....bummer.
